Eastern-Research-Group/mywaterway
# Feature/24 portable map widget

## Related Issues:
* [HMW-24](https://jira.epa.gov/browse/HMW-24)

## Main Changes:
* Added a save panel to the Add data widget, which saves HMW data to ArcGIS Online.

## Steps To Test:
1. Navigate to http://localhost:3000/community/dc/overview
2. Select layers to save 
3. Click save and login
4. Go to your  AGO account and verify you have the expected output

## TODO:
I'll start working on this stuff, but I'm ok with merging this in without completing any of the below items. If I don't get to these I'll create separate tickets for these items.

- [x] Add publishing logic for the Upstream Watershed layer
- [x] Add publishing logic for the Identified Issues layer
- [ ] Add publishing logic for the CyAN layer(s)
- [x] Update the `isServiceNameAvailable` function to check for uniqueness in the user's content instead of at the organization level. 
    *  I confirmed in the esri docs that `isServiceNameAvailable`  checks for uniqueness across the organization on not just in the user's content. 
    * I did some more testing on this and esri just doesn't allow duplicate names across the org. So this is item just isn't possible and I'm going to go ahead and mark this one off.
- [ ] Update the HMW generated portion of the description to say what filters have been applied to layers, if any (i.e., dischargers with significant violations, identified issues, etc.).
- [x] ~Fix bug related to saving layers uploaded from a file where they have the same file name.~ Fix bug related to saving multiple layers uploaded from a file where the file name ends with `.zip`.
- [ ] Update popups to match up with HMW more
